src/similaritymatrix/similaritymatrix.cpp

```diff
@@ -0,0 +1,74 @@
+#include "similaritymatrix.h"
+
+#include <iostream>
+#include <string>
+#include <algorithm> 
+
+typedef boost::multi_array<double, 2> array_type;
+typedef array_type::index index;
+typedef boost::array<index, 2> index_tupel;
+
+
+Similarity_Matrix::Similarity_Matrix(const std::string& sequence_x, const std::string& sequence_y) :
+    similarity_matrix(boost::extents[sequence_x.size() + 1][sequence_y.size() + 1]) {}
+
+index_tupel Similarity_Matrix::find_index_of_maximum() const {
+    int distance = std::distance(
+        similarity_matrix.origin(), 
+        std::max_element( 
+            similarity_matrix.origin(), 
+            similarity_matrix.origin() + similarity_matrix.num_elements()
+        )
+    );
+
+    const int dim_y = similarity_matrix.shape()[1];
+    const unsigned int x = distance/dim_y;
+    const unsigned int y = distance - x*dim_y;
+
+    index_tupel idx = {{x, y}};
+
+    std::cout << "Maximum is " << similarity_matrix[x][y] << " @ (" << x << ", " << y << ")" << std::endl;
+
+    return idx;
+}
+
+void Similarity_Matrix::print_matrix() const {
+    index dim_x = similarity_matrix.shape()[0];
+    index dim_y = similarity_matrix.shape()[1];
+
+    for(index x = 0; x < dim_x; x++) {
+        for(index y = 0; y < dim_y; y++) {
+            std::cout << similarity_matrix[x][y] << "\t";
+        }
+        std::cout << "\n";
+    }
+}
+
+void Similarity_Matrix::iterate_anti_diagonal(std::function<double(const array_type& matrix, index local_i, index k)> callback) {
```

Comment:
Iterate in anti-diagonal looks good for parallelization, but I guess it is not very cache efficient. Maybe `similarity_matrix` should be stored in a skewed form (like skyline matrix) to optimize cache locality though it may take twice as much memory.

I'm not sure `callback` will be properly inlined here. Maybe adding `-combine` and `-fwhole-program` could help?  https://stackoverflow.com/questions/8450030/can-modern-c-compilers-inline-functions-that-are-defined-in-a-cpp-file